test/test_ff_utils.py

```diff
@@ -156,6 +156,23 @@ def test_unified_authentication(integrated_ff):
     assert 'Must provide a valid authorization key or ff' in str(exec_info.value)
 
 
+def test_unified_authentication_more_envs():
+    key1 = ff_utils.unified_authentication(ff_env="data")
```

Comment:
I have a concern that this test does not test, especially for the 'staging' and 'data' frameworks, what the credential acquired is for the correct site. My intuition is that it needs to control for all the variabilities that are selecting either fourfront-green or fourfront-blue and make sure that the test succeeds not only in the moment it is run but would also succeed if the stars were aligned differently and the swap was the other way. Sometimes otherwise tests like this mysteriously start failing when the big switch is thrown (cname swap). So I think the items tested here are an improvement because they do test that something came back under some circumstances, but they aren't the full set of things we should be testing. I'm willing to look at how that might be done but not today. Since this is test-only stuff, it can presumably sit in a holding pattern for a few days until we've had a chance to work through this.

Reply:
Well, keys are same for data and staging, so the end result should be the same, though the environment it has to look up is different. I made some edits that ensures it finds the same keys for data/staging/green/blue but as to whether the correct env is found, I assume we would need to test for that on whatever function actually looks that up. I haven't had time to look into whether those tests are there yet.

Reply:
I checked that s3Utils has code for getting the common prod bucket right now while these are shared buckets. We may need to change the implementation at some point, to make the credentials be different, but I think this test would notice that change and we'd end up having to write better tests at that time. So I think this is OK.